test/e2e/txsimNode.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	txsimDockerSrcURL = "ghcr.io/celestiaorg/txsim"
+)
+
+func txsimDockerImageName(version string) string {
+	return fmt.Sprintf("%s:%s", txsimDockerSrcURL, version)
+}
+
+type TxSim struct {
+	Name     string
+	Instance *knuu.Instance
+}
+
+func CreateTxSimNode(
+	name, version string,
+	endpoint string,
+	seed int,
+	sequences int,
+	blobRange string,
+	pollTime int,
+	resources Resources,
+	volumePath string,
+) (*TxSim, error) {
+	instance, err := knuu.NewInstance(name)
+	if err != nil {
+		return nil, err
+	}
+	image := txsimDockerImageName(version)
+	log.Info().
+		Str("name", name).
+		Str("image", image).
+		Msg("setting image for txsim node")
+	err = instance.SetImage(image)
+	if err != nil {
+		log.Err(err).
+			Str("name", name).
+			Str("image", image).
+			Msg("failed to set image for txsim node")
+		return nil, err
+	}
+	err = instance.SetMemory(resources.MemoryRequest, resources.MemoryLimit)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.SetCPU(resources.CPU)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.AddVolumeWithOwner(volumePath, resources.Volume, 10001)
+	if err != nil {
+		return nil, err
+	}
+	args := []string{
+		fmt.Sprintf("-k %d", 0),
+		fmt.Sprintf("-g %s", endpoint),
+		fmt.Sprintf("-t %ds", pollTime),
+		fmt.Sprintf("-b %d ", sequences),
+		fmt.Sprintf("-d %d ", seed),
+		fmt.Sprintf("-a %d ", 5),
+		fmt.Sprintf("-s %s ", blobRange),
```

Comment:
[extremely optional] if txSim supports longer format flags then it could aid readability to use them here. IMO the short form of flags (`git commit -m` can be used on the CLI but the longer format should be used in code / scripts `git commit --message`)

Reply:
The bash script for txsim does not exactly follow the txsim CLI flags naming. Instead, it incorporates its own [set of command-line options](https://github.com/celestiaorg/celestia-app/blob/a0a6e4b8e0d700c80f94e5c7a653ae52199ec8a4/docker/txsim.sh#L18)